shared/signup/phone-number/verify.tsx

```diff
@@ -3,105 +3,101 @@ import * as Kb from '../../common-adapters'
-        titleComponent={
+const VerifyPhoneNumber = (props: Props) => {
+  // Clear data on unmount
+  React.useEffect(() => {
+    return props.onCancel
+  }, [props.onCancel])
+  const [code, onChangeCode] = React.useState('')
+  const disabled = !code
+  const onContinue = () => (disabled ? {} : props.onContinue(code))
+  return (
+    <SignupScreen
+      onBack={props.onBack}
+      banners={props.error ? [<Kb.Banner key="error" color="red" text={props.error} />] : []}
+      buttons={[{label: 'Continue', onClick: onContinue, type: 'Success', waiting: props.verifyWaiting}]}
+      titleComponent={
+        <Kb.Text type="BodyTinySemibold" style={styles.headerText} center={true}>
+          {props.phoneNumber}
+        </Kb.Text>
+      }
+      containerStyle={styles.container}
+      headerStyle={styles.container}
+      header={
+        <Kb.Box2 direction="horizontal" fullWidth={true} alignItems="center" style={styles.headerContainer}>
+          <Kb.Text type="BodyBigLink" style={styles.backButton} onClick={props.onBack}>
+            Back
+          </Kb.Text>
           <Kb.Text type="BodyTinySemibold" style={styles.headerText} center={true}>
-            {this.props.phoneNumber}
+            {props.phoneNumber}
           </Kb.Text>
-        }
-        containerStyle={styles.container}
-        headerStyle={styles.container}
-        header={
-          <Kb.Box2 direction="horizontal" fullWidth={true} alignItems="center" style={styles.headerContainer}>
-            <Kb.Text type="BodyBigLink" style={styles.backButton}>
-              Back
-            </Kb.Text>
-            <Kb.Text type="BodyTinySemibold" style={styles.headerText} center={true}>
-              {this.props.phoneNumber}
-            </Kb.Text>
-          </Kb.Box2>
-        }
-        negativeHeader={true}
-        skipMobileHeader={true}
-      >
-        <Kb.Box2 alignItems="center" direction="vertical" fullWidth={true} gap="small" style={styles.body}>
-          <Kb.Text type="Body" negative={true} center={true}>
-            Enter the code in the SMS you received:
+          <Kb.Box2 direction="horizontal" style={Styles.globalStyles.flexOne} />
+        </Kb.Box2>
+      }
+      negativeHeader={true}
+      skipMobileHeader={true}
+      showHeaderInfoicon={true}
+    >
+      <Kb.Box2 alignItems="center" direction="vertical" fullWidth={true} gap="small" style={styles.body}>
+        <Kb.Text type="Body" negative={true} center={true}>
+          Enter the code in the SMS you received:
+        </Kb.Text>
+        <Kb.PlainInput
+          autoFocus={true}
+          style={styles.input}
+          flexable={true}
+          keyboardType="numeric"
+          onChangeText={onChangeCode}
+          textType="Header"
+          textContentType="oneTimeCode"
+        >
+          <Kb.Text type="Header" style={styles.inputText}>
+            {/* We put this child in Input because some text styles don't work on input itself - the one we need here is letterSpacing */}
```

Comment:
Is that a limitation of the base react/HTML component, or of our `PlainInput`? if the latter, should we modify `PlainInput` to allow this style?

Reply:
It's an RN limitation